### ISSUE TITLE
Insert Null value in Nullable(Int16) Column

### ISSUE THREAD

**Issue by licicego** [killwort/ClickHouse-Net#58]
## Description
Continuing to use your library came across another bug. When I try to insert a NULL value in a Nullable (Int16) column, I get an error.

## Steps to reproduce
1. The TestTable table was created for testing. 

> CREATE TABLE tflow.TestTable (
> `Int16Column` Nullable(Int16),
>  `Name` String
> ) ENGINE = ReplacingMergeTree() ORDER BY Name SETTINGS index_granularity = 32768

2. A test application has been created.

TestTable.cs

    public class TestTable : IEnumerable
    {
        public Int16? Int16Column { get; set; }
        public string Name { get; set; }

        public IEnumerator GetEnumerator()
        {
            yield return Int16Column;
            yield return Name;
        }
    }

--------------------------------------------------
            var testConnection = new  ClickHouseConnection(_connString_);

            var list = new List<TestTable>();

            list.Add(new TestTable()
            {
                Int16Column = null,
                Name = "Example 1"
            });

            var commandBulkInsert = testConnection.CreateCommand();
            commandBulkInsert.CommandText = "INSERT INTO TestTable (Int16Column,Name) VALUES @bulk";
            commandBulkInsert.Parameters.Add(new ClickHouseParameter
            {
                ParameterName = "bulk",
                Value = list
            });

            testConnection.Open();
            commandBulkInsert.ExecuteNonQuery();
            testConnection.Close();

**Result**: _System.NullReferenceException: "Object reference not set to an instance of an object."_
-----------------------------
3. For example, the same GUID was inserted using DBeaver. SQL:
`INSERT INTO TestTable VALUES (NULL,'Example 2')`
4. Query result
`SELECT * FROM TestTable`

> Int16Column|Name     |
> ------|---------|
>  [NULL]|Example 2|

**Comment by rquackenbush** [comment]
I'm running into a similar problem with a `Nullable(UUID)` column.

Table:

```sql
CREATE TABLE logs.box_history
(
	id UUID, 
	box_id UUID
) ENGINE = Log
```

```c#
using(var command = new ClickHouseCommand(connection, "insert into box_history (id, box_id) values (@id, @box_id)")
{
    command.Parameters.Add("id", DbType.Guid, Guid.NewGuid());
    command.Parameters.Add("box_id",  null);
}

command.ExecuteNonQuery();
```

Executing this gives me the error:

```
Unhandled Exception: System.InvalidCastException: Cannot convert parameter with type AnsiString to Guid.                   
   at ClickHouse.Ado.Impl.ColumnTypes.GuidColumnType.ValueFromParam(ClickHouseParameter parameter)                         
   at ClickHouse.Ado.Impl.ColumnTypes.NullableColumnType.ValueFromParam(ClickHouseParameter parameter)                    
   at ClickHouse.Ado.ClickHouseCommand.Execute(Boolean readResponse, ClickHouseConnection connection)                      
   at ClickHouse.Ado.ClickHouseCommand.ExecuteNonQuery()                                               
```

Changing the code to:

```c#
command.Parameters.Add("box_id", DbType.Guid, null);
```

Gives this error:

```
Unhandled Exception: System.InvalidCastException: Null object cannot be converted to a value type.
   at System.Convert.ChangeType(Object value, Type conversionType, IFormatProvider provider)
   at ClickHouse.Ado.Impl.ColumnTypes.GuidColumnType.ValueFromParam(ClickHouseParameter parameter)
   at ClickHouse.Ado.Impl.ColumnTypes.NullableColumnType.ValueFromParam(ClickHouseParameter parameter)
   at ClickHouse.Ado.ClickHouseCommand.Execute(Boolean readResponse, ClickHouseConnection connection)
   at ClickHouse.Ado.ClickHouseCommand.ExecuteNonQuery()
```

And changing the code to use DBNull.Value doesn't seem to help either:

```c#
command.Parameters.Add("box_id", DbType.Guid, DBNull.Value);
```

Gives the error:
```
Unhandled Exception: System.InvalidCastException: Invalid cast from 'System.DBNull' to 'System.Guid'.
   at System.Convert.DefaultToType(IConvertible value, Type targetType, IFormatProvider provider)
   at ClickHouse.Ado.Impl.ColumnTypes.GuidColumnType.ValueFromParam(ClickHouseParameter parameter)
   at ClickHouse.Ado.Impl.ColumnTypes.NullableColumnType.ValueFromParam(ClickHouseParameter parameter)
   at ClickHouse.Ado.ClickHouseCommand.Execute(Boolean readResponse, ClickHouseConnection connection)
   at ClickHouse.Ado.ClickHouseCommand.ExecuteNonQuery()
```

How do I get null values into the database?

**Comment by killwort** [comment]
I've merged #61 hope that will work.
